--- configure.ac	2012-03-15 20:50:51.871484426 +0000
+++ configure.ac	2012-03-15 20:51:33.645962171 +0000
@@ -1855,14 +1855,6 @@
     fi
 }
 
-gallium_require_llvm() {
-    if test "x$MESA_LLVM" = x0; then
-        case "$host_cpu" in
-        i*86|x86_64) AC_MSG_ERROR([LLVM is required to build $1 on x86 and x86_64]);;
-        esac
-    fi
-}
-
 dnl Gallium drivers
 dnl Duplicates in GALLIUM_DRIVERS_DIRS are removed by sorting it after this block
 if test "x$with_gallium_drivers" != x; then
@@ -1884,7 +1876,6 @@
             ;;
         xr300)
             PKG_CHECK_MODULES([RADEON], [libdrm_radeon >= $LIBDRM_RADEON_REQUIRED])
-            gallium_require_llvm "Gallium R300"
             GALLIUM_DRIVERS_DIRS="$GALLIUM_DRIVERS_DIRS r300"
             gallium_check_st "radeon/drm" "dri-r300" "xorg-r300" "" "xvmc-r300" "vdpau-r300" "va-r300"
             ;;
