diff --git a/configure.ac b/configure.ac
index 21a1986..94b2247 100644
@@ -1891,7 +1891,6 @@
--- a/configure.ac
+++ b/configure.ac
             PKG_CHECK_MODULES([RADEON], [libdrm_radeon >= $LIBDRM_RADEON_REQUIRED])
             gallium_require_drm "Gallium R300"
             gallium_require_drm_loader
-            gallium_require_llvm "Gallium R300"
             ;;
         xr600)
             HAVE_GALLIUM_R600=yes
