--- include/linux/aufs_type.h	2012-06-28 17:55:10.205661800 +0200
+++ include/linux/aufs_type.h	2012-06-18 19:22:37.000000000 +0200
@@ -21,21 +21,8 @@
 
 #define AUFS_NAME	"aufs"
 
-#ifdef __KERNEL__
-/*
- * define it before including all other headers.
- * sched.h may use pr_* macros before defining "current", so define the
- * no-current version first, and re-define later.
- */
-#define pr_fmt(fmt)	AUFS_NAME " %s:%d: " fmt, __func__, __LINE__
-#include <linux/sched.h>
-#undef pr_fmt
-#define pr_fmt(fmt)	AUFS_NAME " %s:%d:%s[%d]: " fmt, \
-		__func__, __LINE__, current->comm, current->pid
-#else
 #include <stdint.h>
 #include <sys/types.h>
-#endif /* __KERNEL__ */
 
 #include <linux/limits.h>
 
@@ -60,11 +47,6 @@
 #endif
 #endif
 
-#ifdef __KERNEL__
-#ifndef AUFS_BRANCH_MAX
-#error unknown CONFIG_AUFS_BRANCH_MAX value
-#endif
-#endif /* __KERNEL__ */
 
 /* ---------------------------------------------------------------------- */
 
@@ -145,11 +127,6 @@
 #define __aligned(x)			__attribute__((aligned(x)))
 #endif
 
-#ifdef __KERNEL__
-#ifndef __packed
-#define __packed			__attribute__((packed))
-#endif
-#endif
 
 struct au_rdu_cookie {
 	uint64_t	h_pos;
@@ -168,7 +145,7 @@
 	char		name[0];
 } __aligned(8);
 
-static inline int au_rdu_len(int nlen)
+static __inline__ int au_rdu_len(int nlen)
 {
 	/* include the terminating NULL */
 	return ALIGN(sizeof(struct au_rdu_ent) + nlen + 1,
@@ -176,7 +153,7 @@
 }
 
 union au_rdu_ent_ul {
-	struct au_rdu_ent 	*e;
+	struct au_rdu_ent *e;
 	uint64_t			ul;
 };
 
