--- usr/share/check_mk/agents/plugins/mk_mysql	2016-12-01 00:08:31.000000000 +0100
+++ usr/share/check_mk/agents/plugins/mk_mysql	2016-12-07 11:02:10.864957366 +0100
@@ -24,11 +24,11 @@
 # to the Free Software Foundation, Inc., 51 Franklin St,  Fifth Floor,
 # Boston, MA 02110-1301 USA.
 
-if which mysqladmin > /dev/null
-then
+mysqladmin="$(command -v mysqladmin 2>/dev/null)"
+if [ -n "$mysqladmin" ]; then
   # Check if mysqld is running and root password setup
   echo "<<<mysql_ping>>>"
-  mysqladmin --defaults-extra-file=$MK_CONFDIR/mysql.cfg ping 2>&1
+  $mysqladmin --defaults-extra-file=$MK_CONFDIR/mysql.cfg ping 2>&1
 
   mysql --defaults-extra-file=$MK_CONFDIR/mysql.cfg -sN \
      -e "select '<<<mysql>>>' ; show global status; show global variables; select '<<<mysql_capacity>>>' ; SELECT table_schema, sum( data_length + index_length ), sum( data_free )  FROM information_schema.TABLES GROUP BY table_schema;"
