--- a/kernel/drv/oss_audigyls/oss_audigyls.c	Thu Jan 06 07:55:36 2011 +0200
+++ b/kernel/drv/oss_audigyls/oss_audigyls.c	Tue Feb 15 04:54:33 2011 +0200
@@ -1648,7 +1648,7 @@
 int
 oss_audigyls_attach (oss_device_t * osdev)
 {
-  int tmp, err, i, tries;
+  int tmp, err, i;
   unsigned char pci_irq_line, pci_revision;
   unsigned short pci_command, vendor, device;
   unsigned int pci_ioaddr;
@@ -1796,15 +1796,9 @@
       /* for SBLive 7.1 */
       OUTL (devc->osdev, 0x005f4301, devc->base + 0x18);
       audigyls_i2c_write (devc, 0x15, 0x4);
-      tries = 0;
-    again:
-      for (i = 0; i < sizeof (spi_dac); i++)
+      for (i = 0; i < (sizeof (spi_dac)/sizeof (spi_dac[0])); i++)
 	{
-	  if (!audigyls_spi_write (devc, spi_dac[i]) && tries < 100)
-	    {
-	      tries++;
-	      goto again;
-	    }
+	  audigyls_spi_write (devc, spi_dac[i]);
 	}
     }
 

--- a/kernel/framework/sndstat/oss_sndstat.c	Thu Jan 06 07:55:36 2011 +0200
+++ b/kernel/framework/sndstat/oss_sndstat.c	Tue Feb 15 04:54:33 2011 +0200
@@ -65,7 +65,7 @@
 
   put_status ("OSS " OSS_VERSION_STRING);
   put_status (oss_license_string);
-  put_status (" (C) 4Front Technologies 1996-2009\n");
+  put_status (" (C) 4Front Technologies 1996-2011\n");
 
   if (riptide_notice)
     put_status ("RipTide Driver (C) 2000, Conexant Systems, Inc.\n");

--- a/setup/FreeBSD/pkg-message	Thu Jan 06 07:55:36 2011 +0200
+++ b/setup/FreeBSD/pkg-message	Tue Feb 15 04:54:33 2011 +0200
@@ -78,9 +78,9 @@
 
 ______________________________________________________________________________
 
-4Front Technologies
-4035 Lafayette Place Unit F
-Culver City, CA 90232
 ront Technologies, Inc.
+9826 Beverlywood Street
+Los Angeles, CA 90064
 U.S.A.
 
 Tel: (+01) 310-202-8530         WWW: http://www.opensound.com

--- a/setup/SunOS/confgen.c	Thu Jan 06 07:55:36 2011 +0200
+++ b/setup/SunOS/confgen.c	Tue Feb 15 04:54:33 2011 +0200
@@ -178,9 +178,7 @@
   else
     fprintf (conf, "name=\"%s\" parent=\"pseudo\" instance=0 ", module);
 
-#ifdef SOL9
    fprintf (conf, "ddi-no-autodetach=1 ddi-forceattach=1 \n");
-#endif
 
   sprintf (tmp, "%s/.params", srcdir);
   if ((f = fopen (tmp, "r")) != NULL)

