--- a/quickreference/Makefile.am	2012-12-13 14:39:44.354054850 +0100
+++ b/quickreference/Makefile.am	2012-12-13 14:40:42.450328544 +0100
@@ -2,7 +2,7 @@
 
 ALL_LINGUAS ?= de el en es fr it ja ko nl nn pl ru sl sv zh_CN
 QUICKREFERENCE_ALL_LINGUAS ?= de el en fi fr it ja ko ru sl sv zh_CN
-QUICKREFERENCE_LINGUAS = $(filter $(ALL_LINGUAS), $(QUICKREFERENCE_ALL_LINGUAS))
+QUICKREFERENCE_LINGUAS = $(filter $(QUICKREFERENCE_ALL_LINGUAS), $(LINGUAS))
 
 PO_FILES = \
 	po/gimp-keys.pot\
--- a/quickreference/Makefile.in	2012-12-13 14:39:48.883998216 +0100
+++ b/quickreference/Makefile.in	2012-12-13 14:42:12.766199439 +0100
@@ -148,7 +148,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-QUICKREFERENCE_LINGUAS = $(filter $(ALL_LINGUAS), $(QUICKREFERENCE_ALL_LINGUAS))
+QUICKREFERENCE_LINGUAS = $(filter $(QUICKREFERENCE_ALL_LINGUAS), $(LINGUAS))
 PO_FILES = \
 	po/gimp-keys.pot\
 	po/de.po	\
