
--- plugins/arpeggiator/arpeggiator.cpp	2014-10-23 11:29:38.817024826 +0200
+++ plugins/arpeggiator/arpeggiator.cpp	2014-10-23 11:30:14.399024938 +0200
@@ -23,8 +23,7 @@
 
 #include <cstring>
 
-#include <lv2plugin.hpp>
-// XXX - Really shouldn't rely on internal headers from lv2-c++-tools
+#include <lv2-c++-tools/lv2plugin.hpp>
 #include <lv2-c++-tools/lv2_event_helpers.h>
 
 
--- plugins/beep/beep.cpp	2014-10-23 11:29:38.818024826 +0200
+++ plugins/beep/beep.cpp	2014-10-23 11:30:36.389025007 +0200
@@ -27,8 +27,7 @@
 #include <fstream>
 #include <string>
 
-#include <lv2synth.hpp>
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
+#include <lv2-c++-tools/lv2synth.hpp>
 #include <lv2-c++-tools/lv2_osc.h>
 
 #include "beep.peg"
--- plugins/beep/beep_gtk.cpp	2014-10-23 11:29:38.818024826 +0200
+++ plugins/beep/beep_gtk.cpp	2014-10-23 11:30:53.188025060 +0200
@@ -24,8 +24,7 @@
 #include <iostream>
 
 #include <gtkmm.h>
-#include <lv2gui.hpp>
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
+#include <lv2-c++-tools/lv2gui.hpp>
 #include <lv2-c++-tools/lv2_event_helpers.h>
 
 #include "beep.peg"
--- plugins/control2midi/control2midi.cpp	2014-10-23 11:29:38.817024826 +0200
+++ plugins/control2midi/control2midi.cpp	2014-10-23 11:31:15.025025129 +0200
@@ -21,8 +21,7 @@
 
 ****************************************************************************/
 
-#include <lv2plugin.hpp>
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
+#include <lv2-c++-tools/lv2plugin.hpp>
 #include <lv2-c++-tools/lv2_event_helpers.h>
 
 #include "control2midi.peg"
--- plugins/envelope/envelope.cpp	2014-10-23 11:29:38.666024825 +0200
+++ plugins/envelope/envelope.cpp	2014-10-23 11:31:34.387025190 +0200
@@ -25,7 +25,7 @@
 #include <cstring>
 #include <iostream>
 
-#include <lv2plugin.hpp>
+#include <lv2-c++-tools/lv2plugin.hpp>
 
 #include "envelope.hpp"
 #include "envelope.peg"
--- plugins/envelope/envelope_gtk.cpp	2014-10-23 11:29:38.667024825 +0200
+++ plugins/envelope/envelope_gtk.cpp	2014-10-23 11:32:12.878025311 +0200
@@ -26,7 +26,7 @@
 
 #include <gtkmm.h>
 
-#include "lv2gui.hpp"
+#include <lv2-c++-tools/lv2gui.hpp>
 #include "envelopeeditor.hpp"
 
 
--- plugins/klaviatur/klaviatur.cpp	2014-10-23 11:29:38.818024826 +0200
+++ plugins/klaviatur/klaviatur.cpp	2014-10-23 11:32:31.978025372 +0200
@@ -23,9 +23,8 @@
 
 #include <iomanip>
 #include <iostream>
-#include <lv2plugin.hpp>
+#include <lv2-c++-tools/lv2plugin.hpp>
 #include "klaviatur.peg"
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
 #include <lv2-c++-tools/lv2_event_helpers.h>
 
 
--- plugins/math-constants/math-constants.cpp	2014-10-23 11:29:38.668024825 +0200
+++ plugins/math-constants/math-constants.cpp	2014-10-23 11:33:06.172025479 +0200
@@ -25,7 +25,7 @@
 #include <cmath>
 #include <iostream>
 
-#include "lv2plugin.hpp"
+#include <lv2-c++-tools/lv2plugin.hpp>
 
 
 namespace {
--- plugins/math-functions/math-functions.cpp	2014-10-23 11:29:38.668024825 +0200
+++ plugins/math-functions/math-functions.cpp	2014-10-23 11:33:26.707025544 +0200
@@ -25,7 +25,7 @@
 #include <cmath>
 #include <iostream>
 
-#include "lv2plugin.hpp"
+#include <lv2-c++-tools/lv2plugin.hpp>
 
 
 using namespace std;
--- plugins/peakmeter/peakmeter.cpp	2014-10-23 11:29:38.668024825 +0200
+++ plugins/peakmeter/peakmeter.cpp	2014-10-23 11:34:01.978025655 +0200
@@ -23,7 +23,7 @@
 
 #include <cmath>
 
-#include "lv2plugin.hpp"
+#include <lv2-c++-tools/lv2plugin.hpp>
 #include "peakmeter.peg"
 
 
--- plugins/rudolf556/rudolf556.cpp	2014-10-23 11:29:38.670024825 +0200
+++ plugins/rudolf556/rudolf556.cpp	2014-10-23 11:35:17.578025894 +0200
@@ -27,7 +27,7 @@
 #include <samplerate.h>
 #include <sndfile.h>
 
-#include "lv2synth.hpp"
+#include <lv2-c++-tools/lv2synth.hpp>
 #include "rudolf556.peg"
 
 
--- plugins/rudolf556/rudolf556_gtk.cpp	2014-10-23 11:29:38.670024825 +0200
+++ plugins/rudolf556/rudolf556_gtk.cpp	2014-10-23 11:35:00.942025841 +0200
@@ -25,7 +25,7 @@
 #include <string>
 
 #include <gtkmm.h>
-#include <lv2gui.hpp>
+#include <lv2-c++-tools/lv2gui.hpp>
 
 #include "rudolf556widget.hpp"
 #include "rudolf556.peg"
