--- plugins/sineshaper/sineshaper.hpp	2014-10-23 22:27:00.987040930 +0200
+++ plugins/sineshaper/sineshaper.hpp	2014-10-23 22:38:51.634043170 +0200
@@ -23,7 +23,7 @@
 #ifndef SINESHAPER_HPP
 #define SINESHAPER_HPP
 
-#include "lv2plugin.hpp"
+#include <lv2plugin.hpp>
 #include "sineoscillator.hpp"
 #include "adsr.hpp"
 #include "frequencytable.hpp"
--- plugins/arpeggiator/arpeggiator.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/arpeggiator/arpeggiator.cpp	2014-10-23 22:33:53.410042230 +0200
@@ -24,8 +24,7 @@
 #include <cstring>
 
 #include <lv2plugin.hpp>
-// XXX - Really shouldn't rely on internal headers from lv2-c++-tools
-#include <lv2cxx_common/lv2_event_helpers.h>
+#include <lv2_event_helpers.h>
 
 
 using namespace std;
--- plugins/beep/beep.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/beep/beep.cpp	2014-10-23 22:34:39.426042375 +0200
@@ -28,8 +28,7 @@
 #include <string>
 
 #include <lv2synth.hpp>
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
-#include <lv2cxx_common/lv2_osc.h>
+#include <lv2_osc.h>
 
 #include "beep.peg"
 
--- plugins/beep/beep_gtk.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/beep/beep_gtk.cpp	2014-10-23 22:34:16.104042302 +0200
@@ -25,8 +25,7 @@
 
 #include <gtkmm.h>
 #include <lv2gui.hpp>
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
-#include <lv2cxx_common/lv2_event_helpers.h>
+#include <lv2_event_helpers.h>
 
 #include "beep.peg"
 
--- plugins/control2midi/control2midi.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/control2midi/control2midi.cpp	2014-10-23 22:34:55.252042425 +0200
@@ -22,8 +22,7 @@
 ****************************************************************************/
 
 #include <lv2plugin.hpp>
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
-#include <lv2cxx_common/lv2_event_helpers.h>
+#include <lv2_event_helpers.h>
 
 #include "control2midi.peg"
 
--- plugins/envelope/envelope_gtk.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/envelope/envelope_gtk.cpp	2014-10-23 22:35:16.516042492 +0200
@@ -26,7 +26,7 @@
 
 #include <gtkmm.h>
 
-#include "lv2gui.hpp"
+#include <lv2gui.hpp>
 #include "envelopeeditor.hpp"
 
 
--- plugins/klaviatur/klaviatur.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/klaviatur/klaviatur.cpp	2014-10-23 22:35:47.281042589 +0200
@@ -25,8 +25,7 @@
 #include <iostream>
 #include <lv2plugin.hpp>
 #include "klaviatur.peg"
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
-#include <lv2cxx_common/lv2_event_helpers.h>
+#include <lv2_event_helpers.h>
 
 
 using namespace LV2;
--- plugins/math-constants/math-constants.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/math-constants/math-constants.cpp	2014-10-23 22:36:08.445042656 +0200
@@ -25,7 +25,7 @@
 #include <cmath>
 #include <iostream>
 
-#include "lv2plugin.hpp"
+#include <lv2plugin.hpp>
 
 
 namespace {
--- plugins/math-functions/math-functions.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/math-functions/math-functions.cpp	2014-10-23 22:36:24.936042708 +0200
@@ -25,7 +25,7 @@
 #include <cmath>
 #include <iostream>
 
-#include "lv2plugin.hpp"
+#include <lv2plugin.hpp>
 
 
 using namespace std;
--- plugins/peakmeter/peakmeter.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/peakmeter/peakmeter.cpp	2014-10-23 22:36:50.332042788 +0200
@@ -23,7 +23,7 @@
 
 #include <cmath>
 
-#include "lv2plugin.hpp"
+#include <lv2plugin.hpp>
 #include "peakmeter.peg"
 
 
--- plugins/peakmeter/peakmeter_gtk.cpp	2014-10-23 22:27:00.986040930 +0200
+++ plugins/peakmeter/peakmeter_gtk.cpp	2014-10-23 22:36:40.128042756 +0200
@@ -25,7 +25,7 @@
 
 #include <gtkmm.h>
 
-#include "lv2gtkgui.hpp"
+#include <lv2gtkgui.hpp>
 #include "vuwidget.hpp"
 
 
--- plugins/rudolf556/rudolf556.cpp	2014-10-23 22:27:00.987040930 +0200
+++ plugins/rudolf556/rudolf556.cpp	2014-10-23 22:37:13.984042862 +0200
@@ -27,7 +27,7 @@
 #include <samplerate.h>
 #include <sndfile.h>
 
-#include "lv2synth.hpp"
+#include <lv2synth.hpp>
 #include "rudolf556.peg"
 
 
--- plugins/sineshaper/sineshaper.cpp	2014-10-23 22:27:00.987040930 +0200
+++ plugins/sineshaper/sineshaper.cpp	2014-10-23 22:38:16.814043060 +0200
@@ -23,8 +23,7 @@
 #include <cmath>
 #include <iostream>
 
-// XXX - Really shouldn't rely on internal headers of lv2-c++-tools
-#include <lv2cxx_common/lv2_event_helpers.h>
+#include <lv2_event_helpers.h>
 
 #include "sineshaper.hpp"
 #include "sineshaperports.hpp"
--- plugins/sineshaper/sineshaper_gtk.cpp	2014-10-23 22:27:00.987040930 +0200
+++ plugins/sineshaper/sineshaper_gtk.cpp	2014-10-23 22:38:27.784043095 +0200
@@ -25,7 +25,7 @@
 
 #include <gtkmm.h>
 
-#include "lv2gtkgui.hpp"
+#include <lv2gtkgui.hpp>
 #include "sineshaperwidget.hpp"
 #include "sineshaper.peg"
 
