--- a/Makefile	2015-08-05 10:31:47.712994203 +0200
+++ b/Makefile	2015-08-05 10:35:12.288992694 +0200
@@ -1,4 +1,4 @@
-CPP    := gcc -Wall -g -fPIC -std=c99
+CPP   := gcc -Wall -g -fPIC -std=c99
 LIB   := 
 INC   := -I. -Ios/ -I/usr/include -I/usr/local/include 
 C_SRCS:= $(wildcard *.c) $(wildcard os/*.c)
@@ -54,18 +54,18 @@
 	$(CPP) -o $(PSTAT) $(TOOL_OBJ) $(LIBINC) -L. -lpstat
 
 libpstat-install: libpstat $(PC_FILE)
-	mkdir -p $(LIBDIR) $(PKGCONFIGDIR)
-	cp -a $(LIBPSTAT) $(LIBPSTAT_SO) $(LIBPSTAT_LIB) $(LIBDIR)
-	cp -a $(PC_FILE) $(PKGCONFIGDIR)
+	mkdir -p $(DESTDIR)$(LIBDIR) $(DESTDIR)$(PKGCONFIGDIR)
+	cp -a $(LIBPSTAT) $(LIBPSTAT_SO) $(LIBPSTAT_LIB) $(DESTDIR)$(LIBDIR)
+	cp -a $(PC_FILE) $(DESTDIR)$(PKGCONFIGDIR)
 
 pstat-install: pstat
-	mkdir -p $(BINDIR)
-	cp -a $(PSTAT) $(BINDIR)
+	mkdir -p $(DESTDIR)$(BINDIR)
+	cp -a $(PSTAT) $(DESTDIR)$(BINDIR)
 
 libpstat-dev-install: libpstat
-	mkdir -p $(INCLUDEDIR) $(INCLUDEDIR)/os
-	cp -a $(wildcard os/*.h) $(INCLUDEDIR)/os
-	cp -a $(wildcard *.h) $(INCLUDEDIR)
+	mkdir -p $(DESTDIR)$(INCLUDEDIR) $(DESTDIR)$(INCLUDEDIR)/os
+	cp -a $(wildcard os/*.h) $(DESTDIR)$(INCLUDEDIR)/os
+	cp -a $(wildcard *.h) $(DESTDIR)$(INCLUDEDIR)
 
 install: libpstat-install pstat-install libpstat-dev-install
 
