--- a/printui/src/keytext.c	2012-05-28 05:07:47.032802136 +0200
+++ b/printui/src/keytext.c	2012-05-28 05:08:01.751618126 +0200
@@ -38,7 +38,7 @@
 #	include <glade/glade.h>
 #endif
 /* #include <gnome-xml/parser.h> */
-#include <libxml/parser.h>	/* Ver.2.80 */
+#include <libxml2/parser.h>	/* Ver.2.80 */
 #include <string.h>
 
 #ifndef	USE_LIB_GLADE
--- a/cngpijmon/src/Makefile.am	2012-05-28 05:04:09.303524126 +0200
+++ b/cngpijmon/src/Makefile.am	2012-05-28 05:06:41.364623101 +0200
@@ -19,6 +19,6 @@
 	interface.c interface.h support.c support.h 
 
 
-#cngpijmon_LDADD = @GTK_LIBS@ $(INTLLIBS) -lxml -lcups 
-cngpijmon_LDADD = @GTK_LIBS@ $(INTLLIBS) -lcups 
+cngpijmon_LDADD = @GTK_LIBS@ $(INTLLIBS) -lxml2 -lcups 
+#cngpijmon_LDADD = @GTK_LIBS@ $(INTLLIBS) -lcups 
 
