--- a/cngpij/cngpij/Makefile.am	2012-05-25 19:27:28.110087679 +0200
+++ b/cngpij/cngpij/Makefile.am	2012-05-25 19:27:49.190824133 +0200
@@ -4,7 +4,7 @@
 	bjcups.c getipc.c bjutil.c \
 	bjutil.h
 
-cngpij_LDADD= -lpopt -lcups -lparamlist 
+cngpij_LDADD= -lpopt -lcups -lparamlist -ldl
 cngpij_LDFLAGS = -L../../libs/paramlist
 
 AM_CFLAGS= -O2 -Wall -I../include/cncl -I../include/misc -I../../libs/paramlist
