From 3a682dd20df4e978075a1862fcccbbf0ed02f418 Mon Sep 17 00:00:00 2001
From: John Hedges <john@drystone.co.uk>
Date: Tue, 12 Jul 2011 11:40:46 +0200
Subject: [PATCH] Read new settings before the dialog is destroyed

---
 channel.py |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/channel.py b/channel.py
index c0efa92..76cf010 100644
--- a/channel.py
+++ b/channel.py
@@ -871,9 +871,8 @@ class ChannelPropertiesDialog(gtk.Dialog):
 
     def on_response_cb(self, dlg, response_id, *args):
         self.channel.channel_properties_dialog = None
-        self.destroy()
+        name = self.entry_name.get_text()
         if response_id == gtk.RESPONSE_APPLY:
-            name = self.entry_name.get_text()
             self.channel.channel_name = name
             try:
                 self.channel.channel.volume_midi_cc = int(self.entry_volume_cc.get_text())
@@ -883,6 +882,7 @@ class ChannelPropertiesDialog(gtk.Dialog):
                 self.channel.channel.balance_midi_cc = int(self.entry_balance_cc.get_text())
             except ValueError:
                 pass
+        self.destroy()
 
     def on_entry_name_changed(self, entry):
         sensitive = False
@@ -933,9 +933,9 @@ class OutputChannelPropertiesDialog(ChannelPropertiesDialog):
         self.display_solo_buttons.set_active(self.channel.display_solo_buttons)
 
     def on_response_cb(self, dlg, response_id, *args):
-        ChannelPropertiesDialog.on_response_cb(self, dlg, response_id, *args)
         if response_id == gtk.RESPONSE_APPLY:
             self.channel.display_solo_buttons = self.display_solo_buttons.get_active()
+        ChannelPropertiesDialog.on_response_cb(self, dlg, response_id, *args)
 
 
 class NewOutputChannelDialog(OutputChannelPropertiesDialog):
-- 
1.6.5.GIT

