commit 310c10bca8ad5cfad38353ab23d5780c469a1bb2
Author: Jean-Philippe Andre <jp.andre@samsung.com>
Date:   Fri Oct 24 19:26:55 2014 +0900

    Really fix the compilation with wayland
    
    ... I hope.

diff --git a/src/bin/e_fm/Makefile.mk b/src/bin/e_fm/Makefile.mk
index 1ab59a4..0eec4f4 100644
--- a/src/bin/e_fm/Makefile.mk
+++ b/src/bin/e_fm/Makefile.mk
@@ -7,6 +7,7 @@ EFM_CPPFLAGS = \
 @cf_cflags@ \
 @VALGRIND_CFLAGS@ \
 @EDJE_DEF@ \
+@WAYLAND_CFLAGS@ \
 -DPACKAGE_BIN_DIR=\"@PACKAGE_BIN_DIR@\" \
 -DPACKAGE_LIB_DIR=\"@PACKAGE_LIB_DIR@\" \
 -DPACKAGE_DATA_DIR=\"@PACKAGE_DATA_DIR@\" \
diff --git a/src/modules/Makefile.mk b/src/modules/Makefile.mk
index 5b6bf8b..348b87f 100644
--- a/src/modules/Makefile.mk
+++ b/src/modules/Makefile.mk
@@ -6,6 +6,7 @@ MOD_CPPFLAGS = -I. \
 -I$(top_builddir)/src/bin \
 -I$(top_srcdir)/src/modules \
 @e_cflags@ \
+@WAYLAND_CFLAGS@ \
 -DE_BINDIR=\"$(bindir)\"
 
 MOD_LIBS = @e_libs@ @dlopen_libs@

