--- cmd/osspartysh/osspartysh.c.old	2009-03-23 23:47:17.000000000 +0200
+++ cmd/osspartysh/osspartysh.c	2009-04-16 14:34:19.000000000 +0300
@@ -344,17 +344,15 @@
 {
 	pid_t pid;
 	int ptyfd;
-	FILE *f;
+	int fd;
 	char tmpl[32]="/tmp/osspartyshXXXXXX";
-	mkstemp(tmpl);
-
-	chmod(tmpl, 0700);
-
-	if ((f=fopen(tmpl, "w"))!=NULL)
-	{
-		fprintf(f, "PS1=\"[SHARED]@\\h:\\w $ \"\n");
-		fprintf(f, "export PS1\n");
-		fclose(f);
+	char* msg = "PS1=\"[SHARED]@\\h:\\w $ \"\nexport PS1\n";
+	
+	if ((fd=mkstemp(tmpl)) != -1)
+	{
+		if(write(fd, msg, strlen(msg)) == -1)
+			perror("write(fd)");
+		close(fd);
 	}
 
 	if ((pid=forkpty(&ptyfd, NULL, NULL, NULL))==-1)
