--- src/libfreebobavc/avc_extended_plug_info.cpp	2008-06-26 12:33:13.000000000 +0200
+++ src/libfreebobavc/avc_extended_plug_info.cpp	2008-06-26 12:34:31.000000000 +0200
@@ -22,6 +22,7 @@
 #include "serialize.h"
 #include "ieee1394service.h"
 
+#include <cstring>
 #include <cstdlib>
 #include <netinet/in.h>
 #include <iostream>
--- src/libfreebobavc/avc_generic.cpp	2008-06-26 12:33:13.000000000 +0200
+++ src/libfreebobavc/avc_generic.cpp	2008-06-26 12:40:19.000000000 +0200
@@ -22,6 +22,7 @@
 #include "serialize.h"
 #include "ieee1394service.h"
 
+#include <cstring>
 #include <cstdlib>
 #include <netinet/in.h>
 
--- tests/test-extplugcmd.cpp	2008-06-26 12:33:13.000000000 +0200
+++ tests/test-extplugcmd.cpp	2008-06-26 12:46:50.000000000 +0200
@@ -24,6 +24,8 @@
 #include "libfreebobavc/ieee1394service.h"
 
 #include <argp.h>
+#include <stdlib.h>
+#include <cstring>
 
 using namespace std;
 
--- tests/test-volume.cpp	2008-06-26 12:33:13.000000000 +0200
+++ tests/test-volume.cpp	2008-06-26 12:47:21.000000000 +0200
@@ -22,6 +22,9 @@
 #include "libfreebobavc/serialize.h"
 #include "libfreebobavc/ieee1394service.h"
 
+#include <stdlib.h>
+#include <cstring>
+
 const bool bVerbose = false;
 
 short int

